aiounifi/models/voucher.py

```diff
@@ -0,0 +1,217 @@
+    start_time: NotRequired[float]
+    end_time: NotRequired[float]
+    for_hotspot: bool
+    admin_name: str
+    status: str
+    status_expires: float
+
+
+@dataclass
+class VoucherListRequest(ApiRequest):
+    """Request object for voucher list."""
+
+    @classmethod
+    def create(cls) -> Self:
+        """Create voucher list request."""
+        return cls(
+            method="get",
+            path="/stat/voucher",
+        )
+
+
+@dataclass
+class VoucherCreateRequest(ApiRequest):
+    """Request object for voucher create."""
+
+    @classmethod
+    def create(
+        cls,
+        number: int,
+        quota: int,
+        expire_number: int,
+        expire_unit: int = 1,
+        usage_quota: int | None = None,
+        rate_max_up: int | None = None,
+        rate_max_down: int | None = None,
+        note: str | None = None,
+    ) -> Self:
+        """Create voucher create request.
+
+        :param number: number of vouchers
+        :param quota: number of using; 0 = unlimited
+        :param expire_number: expiration of voucher per expire_unit
+        :param expire_unit: scale of expire_number, 1 = minute, 60 = hour, 3600 = day
+        :param usage_quota: quantity of bytes allowed in MB
+        :param rate_max_up: up speed allowed in kbps
+        :param rate_max_down: down speed allowed in kbps
+        :param note: description
+        """
+        data = {
+            "cmd": "create-voucher",
+            "n": number,
+            "quota": quota,
+            "expire_number": expire_number,
+            "expire_unit": expire_unit,
+        }
+        if usage_quota:
+            data["bytes"] = usage_quota
+        if rate_max_up:
+            data["up"] = rate_max_up
+        if rate_max_down:
+            data["down"] = rate_max_down
+        if note:
+            data["note"] = note
+
+        return cls(
+            method="post",
+            path="/cmd/hotspot",
+            data=data,
+        )
+
+
+@dataclass
+class VoucherDeleteRequest(ApiRequest):
+    """Request object for voucher delete."""
+
+    @classmethod
+    def create(
+        cls,
+        obj_id: str,
+    ) -> Self:
+        """Create voucher delete request."""
+        data = {
+            "cmd": "delete-voucher",
+            "_id": obj_id,
+        }
+        return cls(
+            method="post",
+            path="/cmd/hotspot",
+            data=data,
+        )
+
+
+class Voucher(ApiItem):
+    """Represents a voucher."""
+
+    raw: TypedVoucher
+
+    @property
+    def id(self) -> str:
+        """ID of voucher."""
+        return self.raw["_id"]
+
+    @property
+    def site_id(self) -> str:
+        """Site ID."""
+        return self.raw["_id"]
+
+    @property
+    def note(self) -> str:
+        """Note describing voucher."""
+        return self.raw.get("note") or ""
+
+    @property
+    def code(self) -> str:
+        """Code in known format 00000-00000."""
+        if len(c := self.raw.get("code", "")) > 5:
```

Comment:
```suggestion
        if len(c := str(self.raw.get("code", ""))) > 5:
```
In your voucher data code can be an int